### PR TITLE
algorithm: Liouville lambda function

### PR DESCRIPTION
**Pull request by itsAkshayDubey**
This PR contains commits related to the implementation of [Liouville lambda function](https://en.wikipedia.org/wiki/Liouville_function). Kindly have a look.

### CODE REVIEW

File: math/liouville.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"errors"
+
+	"github.com/TheAlgorithms/Go/math/prime"
+)
+
+var ErrNonZeroArgsOnly error = errors.New("arguments must be greater than zero")
+
+// Lambda is the liouville function
+// This function returns λ(n) for given number
+func Lambda(n int) (int, error) {
```
Comment (tjgurwara99) [inline math/liouville.go:24]: There are many things in CS that are called `Lambda`. I think you should make this `LiouvilleLambda` or `LiouvilleFunc` to make it easier to not get this confused with other `Lambda`s.

File: math/liouville.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	if n < 0 {
+		return 0, ErrPosArgsOnly
+	}
+	if n == 0 {
+		return 0, ErrNonZeroArgsOnly
+	}
+	if len(prime.Factorize(int64(n)))%2 == 0 {
+		return 1, nil
+	}
+	return -1, nil
```
Comment (tjgurwara99) [inline math/liouville.go:34]: When you see more than 2 `if` statements, consider whether you can use a `switch` case instead. This is mostly styling preference and therefore not blocking.

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: math/liouville.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+package math
+
+import (
+	"errors"
+
+	"github.com/TheAlgorithms/Go/math/prime"
+)
+
+var ErrNonZeroArgsOnly error = errors.New("arguments must be greater than zero")
```
Comment (tjgurwara99) [inline math/liouville.go:20]: The name is not correct. The name of the error implies that the arguments also accepts negative number but the comment inside the error indicates that only positive numbers are allowed.

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

**Review by tjgurwara99 — APPROVED** [review]

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**